Code/Tools/AssetProcessor/native/utilities/UuidManager.h

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <AzCore/IO/Path/Path.h>
+#include <AzCore/Interface/Interface.h>
+#include <AzCore/std/containers/unordered_set.h>
+#include <AzCore/std/string/string.h>
+#include <native/AssetManager/SourceAssetReference.h>
+
+namespace AzToolsFramework
+{
+    struct IMetadataRequests;
+}
+
+namespace AssetProcessor
+{
+    struct IUuidRequests
+    {
+        AZ_RTTI(IUuidRequests, "{4EA7E0F6-CB4E-4F9C-ADBC-807676D51772}");
+
+        virtual ~IUuidRequests() = default;
+
+        //! Gets the canonical UUID for a given source asset.  A new metadata file may be created with a randomly generated UUID if
+        //! generation is enabled for the file type.  Otherwise the UUID returned will follow the legacy system based on the file path.
+        virtual AZ::Uuid GetUuid(const SourceAssetReference& sourceAsset) = 0;
+        //! Gets the set of legacy UUIDs for the given source asset.  These are all the UUIDs that may have been used to reference this asset based on previous generation methods.
```

Comment:
does "legacy UUIDs" equal "previous UUIDs"? If so, can the term be switched to be previous UUIDs?

Reply:
Maybe? Right now what this means is "the old, deprecated, do-not-generate-new-references-using-this-algorithm set of UUIDs which previously might have been used to refer to this asset".  This is stored in a JSON format that anyone can modify with external tools, or may be updated in the future to include more things, so the list might expand beyond those 2 UUIDs we have currently.

I kind of prefer calling them legacy though because it does imply "these are old and all references using them should move away from them and use the canonical UUID".

Is there a reason to prefer the term "previous"?